test/UnitTests/MSTest.CoreAdapter.Unit.Tests/MSTestSettingsTests.cs

```diff
@@ -1011,6 +1011,21 @@ public void IsLegacyScenarioReturnsTrueWhenTestSettingsFileIsGiven()
             Assert.IsTrue(MSTestSettings.IsLegacyScenario(this.mockMessageLogger.Object));
         }
 
+        [TestMethod]
+        public void IsLegacyScenarioReturnsTrueWhenVsmdiFileIsGiven()
```

Comment:
This test is not required because we don't check for file extension and it is covered in `IsLegacyScenarioReturnsTrueWhenTestSettingsFileIsGiven`.
Consider adding test to check warning message warning message.